convert/convert.go

```diff
@@ -0,0 +1,120 @@
+package convert
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kong/deck/file"
+	"github.com/kong/deck/utils"
+	"github.com/kong/go-kong/kong"
+)
+
+type Format string
+
+const (
+	FormatKongGateway Format = "kong-gateway"
+	FormatKonnect     Format = "konnect"
+)
+
+var (
+	AllFormats = []Format{FormatKongGateway, FormatKonnect}
+
+	conversions = map[Format]map[Format]bool{
+		FormatKongGateway: {
+			FormatKonnect: true,
+		},
+	}
+)
+
+func validConversion(from, to Format) (bool, error) {
+	if fromMap, ok := conversions[from]; ok {
+		if _, ok := fromMap[to]; ok {
+			return ok, nil
+		}
+	}
+
+	return false, fmt.Errorf("cannot convert from '%s' to '%s' format", from, to)
+}
```

Comment:
Comment part 1:

This duplicates the logic of the `switch` in `func Convert()`. Why do we need both? Why not just let `Convert()` fail if no suitable conversion is possible?

Comment part 2 (provided that we need to retain this function for some reason):

Each of the two return values of this function gives exact same information - the `error` additionally giving the error message. This makes the return value space unnecessarily complex: ((true, nil), (true, err), (false, nil), (false, err)).

If you rephrase this function as `func validateConversion(from, to Format) error`, you'll have only one return value. Easier to test and understand, more difficult to use in a buggy way.

Why: imagine that this function somehow ends up returning `false, nil` (because this is a natural way to say "the result evaluated to false with no runtime errors". In such case, the caller (`func Convert`) will erroneously `return nil` (saying "no error").